pkg/operator/ceph/object/admin.go

```diff
@@ -292,6 +295,121 @@ func isInvalidFlagError(err error) bool {
+	// don't expect json output since we don't intend to use the output from the command
+	_, err = runAdminCommand(c, false, "period", "update", "--commit")
+	if err != nil {
+		return errorOrIsNotFound(err, "failed to commit RGW configuration period changes")
+	}
+
+	return nil
+}
+
+// return true if the configuration period will change if the staged period is committed
+func periodWillChange(current, staged string) (bool, error) {
+	// Rook wants to check if there are any differences in the current period versus the period that
+	// is staged to be applied/committed. If there are differences, then Rook should "commit" the
+	// staged period changes to instruct RGWs to update their runtime configuration.
+	//
+	// For many RGW interactions, Rook often creates a typed struct to unmarshal RGW JSON output
+	// into. In those cases Rook is able to opt in to only a small subset of specific fields it
+	// needs. This keeps the coupling between Rook and RGW JSON output as loose as possible while
+	// still being specific enough for Rook to operate.
+	//
+	// For this implementation, we could use a strongly-typed struct here to unmarshal into, and we
+	// could use DisallowUnknownFields() to cause an error if the RGW JSON output changes to flag
+	// when the existing implementation might be invalidated. This relies on an extremely tight
+	// coupling between Rook and the JSON output from RGW. The failure mode of this implementation
+	// is to return an error from the reconcile when there are unmarshalling errors, which results
+	// in CephObjectStores that could not be updated if a version of Ceph changes the RGW output.
+	//
+	// In the chosen implementation, we unmarshal into "dumb" data structures that create a loose
+	// coupling. With these, we must ignore the fields that we have observed to change between the
+	// current and staged periods when we should *not* commit an un-changed period. The failure mode
+	// of this implementation is that if the RGW output changes its structure, Rook may detect
+	// differences where there are none. This would result in Rook committing the period more often
+	// than necessary. Committing the period results in a short period of downtime while RGWs reload
+	// their configuration, but we opt for this inconvenience in lieu of blocking reconciliation.
+	//
+	// For any implementation, if the RGW changes the behavior of its output but not the structure,
+	// Rook could commit unnecessary period changes or fail to commit necessary period changes
+	// depending on how the RGW output has changed. Rook cannot detect this class of failures, and
+	// the behavior cannot be specifically known.
+	var currentJSON map[string]interface{}
+	var stagedJSON map[string]interface{}
+	var err error
+
+	err = json.Unmarshal([]byte(current), &currentJSON)
+	if err != nil {
+		return true, errors.Wrap(err, "failed to unmarshal current RGW configuration period")
```

Comment:
We don't use the output, but this means that if someone forgets to check the error return from this, they would assume changes rather than assume no changes, which is the safer option for users.